Fix: increase wait time in recovery tests to avoid flaking

Anakin100100 opened the pull request:
Currently the tests for recovering after TelemetryCommand transmission failure wait 3 seconds after the initial setup for the telemetry client goroutine to run. This is at the edge of expected wait time for the goroutine to go through all of the required code. Increasing the wait time for five seconds guarantees that by the time that we check the expected logs all of the testing scenario will have already completed. 

Fixes #410

Anakin100100 commented:
It seems like this issue is not related to the wait time of the consumer goroutine. I'm going to try to reproduce in on Windows

aaron-ai commented:
@Anakin100100 Thank you for your contribution! I am not familiar with Golang, and I have a question: does Golang have a tool similar to Java's awaitility that can help us avoid using the sleep method and increase the speed of test execution?

Anakin100100 commented:
@aaron-ai I'm not aware that such tool exists @guyinyou is something like that typical in Go? In this case skipping the sleep period is a bad idea because the function that we are testing is running possibly on another process as a goroutine so there is no guarantee of execution order. This would result in flaking. Running the tests in parallel is a viable idea. I will look into that.

Anakin100100 commented:
I kinda suspect this could be a deadlock somewhere that is hit more often on windows maybe due to different goroutine implementation. I will try to use go 1.17 locally just to make sure.

Anakin100100 commented:
I managed to reproduce the issue only once on a completely fresh installation of Go on my desktop. Subsequent builds and test runs run fine.

Anakin100100 commented:
@aaron-ai could you give me access to run GitHub actions in my pull requests? Currently I can't see if the new changes fix the build issue in cicd. (They seem to work locally)

aaron-ai commented:
> @aaron-ai could you give me access to run GitHub actions in my pull requests? Currently I can't see if the new changes fix the build issue in cicd. (They seem to work locally)

I am trying to investigate the Github Action approval requirement of ASF project, it is a little bit different from other GitHub project.

Anakin100100 commented:
@aaron-ai thanks for pointing out the version change. I think we can merge this now that the tests run fine. To sum up in this pr we:
 - corrected 2 flaky tests
 - fixed flaking on windows in cicd
 - corrected some cosmetic changes pointed out by staticcheck. 

I'm going to work on the data races in #294 next.

aaron-ai commented:
@Anakin100100  Could you comment on this issue https://github.com/apache/rocketmq-clients/issues/294? then I can assign it to you.
